src/base/memory_monitor_cl.cpp

```diff
@@ -0,0 +1,160 @@
+	{
+	  return i;
+	}
+    }
+
+  /* for code protection. unreachable */
+  return -1;
+}
+
+/*
+ * mmon_print_server_info -
+ *
+ * return:
+ *
+ *   server_info(in/out):
+ */
+void
+mmon_print_server_info (MMON_SERVER_INFO &server_info)
+{
+  fprintf (stdout, "====================Cubrid Memmon====================\n");
+  fprintf (stdout, "Server Name: %s\n", server_info.name);
+  fprintf (stdout, "Total Memory Usage(KB): %lu\n\n", CONVERT_TO_KB_SIZE (server_info.total_mem_usage));
+  fprintf (stdout, "-----------------------------------------------------\n");
+}
+
+/*
+ * mmon_print_module_info -
+ *
+ * return:
+ *
+ *   module_info(in):
+ */
+void
+mmon_print_module_info (std::vector<MMON_MODULE_INFO> &module_info)
+{
+  const std::string init_size_str ("Init Size(KB)");
+  const std::string cur_size_str ("Cur Size(KB)");
+  const std::string peak_size_str ("Peak Size(KB)");
+  const std::string resize_expand_str ("Resize Expand");
+
+  for (const auto &m_info : module_info)
+    {
+      fprintf (stdout, "Module Name: %s\n\n", m_info.name);
+      fprintf (stdout, "%-13s\t: %17lu\n", init_size_str.c_str (), CONVERT_TO_KB_SIZE (m_info.stat.init_stat));
+      fprintf (stdout, "%-13s\t: %17lu\n", cur_size_str.c_str (), CONVERT_TO_KB_SIZE (m_info.stat.cur_stat));
+      fprintf (stdout, "%-13s\t: %17lu\n", peak_size_str.c_str (), CONVERT_TO_KB_SIZE (m_info.stat.peak_stat));
+      fprintf (stdout, "%-13s\t: %17u\n\n", resize_expand_str.c_str (), m_info.stat.expand_resize_count);
+
+      for (const auto &comp_info : m_info.comp_info)
+	{
+	  fprintf (stdout, "%s\n", comp_info.name);
+	  fprintf (stdout, "\t%-17s\t: %17lu\n", init_size_str.c_str (), CONVERT_TO_KB_SIZE (comp_info.stat.init_stat));
+	  fprintf (stdout, "\t%-17s\t: %17lu\n", cur_size_str.c_str (), CONVERT_TO_KB_SIZE (comp_info.stat.cur_stat));
+
+	  for (const auto &subcomp_info : comp_info.subcomp_info)
+	    {
+	      std::string out_name (subcomp_info.name);
+	      out_name += "(KB)";
+	      fprintf (stdout, "\t  %-15s\t: %17lu\n", out_name.c_str (), CONVERT_TO_KB_SIZE (subcomp_info.cur_stat));
+	    }
+	  fprintf (stdout, "\t%-17s\t: %17lu\n", peak_size_str.c_str (), CONVERT_TO_KB_SIZE (comp_info.stat.peak_stat));
+	  fprintf (stdout, "\t%-17s\t: %17u\n\n", resize_expand_str.c_str (), comp_info.stat.expand_resize_count);
+	}
+      fprintf (stdout, "\n-----------------------------------------------------\n\n");
+    }
+}
+
+/*
+ * mmon_print_module_info_summary -
+ *
+ * return:
+ *
+ *   module_info(in):
+ */
+void
+mmon_print_module_info_summary (uint64_t server_mem_usage, std::vector<MMON_MODULE_INFO> &module_info)
+{
+  double percent = 0.0;
+
+  fprintf (stdout, "Per Module Memory Usage (KB)\n");
+  fprintf (stdout, "\tModule\t\tModule\t\t\tModule\n");
+  fprintf (stdout, "\tIndex\t\tName\t\t\tUsage(%%)\n");
+
+  for (const auto &m_info : module_info)
+    {
+      if (m_info.stat.cur_stat != 0)
+	{
+	  percent = (double) m_info.stat.cur_stat / (double) server_mem_usage;
```

Comment:
```suggestion
	  percent =  m_info.stat.cur_stat / (double) server_mem_usage;
```

Reply:
Q. `server_mem_usage` is always larger than 0 in any case? even in any error cases?

Reply:
I changed condition to server_mem_usage. This condition check is for devided to 0, but I get a mistake.